cvmfs/repository.py

```diff
@@ -293,6 +293,86 @@ def _retrieve_and_open_catalog(self, catalog_hash):
         self._opened_catalogs[catalog_hash] = new_catalog
         return new_catalog
 
+    def _lookup_path(self, path):
+        """
+        Lookups in the already opened catalogs for this path's best fit
+        :param path: path to search for
+        :return: the DirectoryEntry that corresponds to the given path if
+        it is found in the already loaded catalogs, or None otherwise
+        """
+        if path == '/':
+            path = ''
+        best_fit = self._opened_catalogs_for_path(path)
+        result = best_fit.find_directory_entry(path)
+        while result is None:
+            best_nested = best_fit.find_best_child_for_path(path)
+            if best_nested is None:
+                break
+            best_fit = best_nested.retrieve_from(self)
+            result = best_fit.find_directory_entry(path)
+        return result
+
+    def lookup(self, path):
+        """
+        Lookups up in the repository for a given path
+        :param path: path to search for
+        follow it and return its final representation
+        :return: the DirectoryEntry that corresponds to the given path
+        """
+        result = None
+        index = -1
+        if not path or path == "":
+            path = "/"
+        while index < len(path):
+            index = path.find('/', index + 1)
+            if index < 0:
+                index = len(path)
+                current_path = path
+            else:
+                current_path = path[:index]
+            result = self._lookup_path(current_path)
```

Comment:
this does lookup all directory entries on the way, doesn't it? Why is that necessary?